include/opendht/connstat.h

```diff
@@ -0,0 +1,97 @@
+/*
+ *  Copyright (C) 2014-2020 Savoir-faire Linux Inc.
+ *  Author(s) : Paymon <paymon@savoirfairelinux.com>
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "log.h"
+
+#include <functional>
+#include <thread>
+
+struct nl_sock;
+struct nl_msg;
+
+namespace dht {
+namespace net {
+
+class OPENDHT_PUBLIC ConnectivityStatus
+{
+public:
+    enum class topic : const long unsigned int {
+        NONE_TOPIC = 0,
+        ADDR,
+        ROUTE,
+        LINK,
+        NEIGHT,
+        TOPICS_NEW,
+        NEWLINK,
+        NEWROUTE4,
+        NEWROUTE6,
+        NEWROUTE,
+        NEWADDR4,
+        NEWADDR6,
+        NEWADDR,
+        NEWNEIGH,
+        NEIGHTBL,
+        IPV4_MROUTE,
+        IPV6_MROUTE,
+        IP_MROUTE,
+        TOPICS_DEL,
+        DELLINK,
+        DELROUTE4,
+        DELROUTE6,
+        DELROUTE,
+        DELADDR4,
+        DELADDR6,
+        DELADDR,
+        DELNEIGH,
+        TOPICS_MAX
+    };
+
+    ConnectivityStatus();
+
+    using connstat_event_cb = std::function<void (long unsigned int)>;
```

Comment:
Use CapitalizedCamelCase for type names. Use Topic for the callback type: 

`using ConnectionEventCb = std::function<void(Topic)>;
`